lib/internal/Magento/Framework/Controller/Result/Redirect.php

```diff
@@ -83,6 +84,16 @@ public function setUrl($url)
         return $this;
     }
 
+    /**
+     * URL Getter
+     *
+     * @return string|null
+     */
+    public function getUrl()
```

Comment:
Unfortunately, adding new public methods to classes is backward incompatible according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.4/contributor-guide/backward-compatible-development/)

Reply:
@sidolov 
Ok, but what would be the best/correct approach for the described use case: https://github.com/magento/magento2/pull/28293#issue-420587913 ?

Reply:
@sidolov Please point out where in the BC development guide this is stated. I don't find it.

Reply:
@schmengler you may find the information in the section: **Introduction of a method to a class or interface**
> Create a new interface with a new method instead of introducing a method to a class or interface.

of https://devdocs.magento.com/contributor-guide/backward-compatible-development/

Reply:
But in this case it isn't productive to introduce a new class. The `setUrl()` method isn't even part of an interface, neither the newly introduced `getUrl()`.

Let's think about the reasoning behind the rule: Adding a method to an interface breaks alternative implementations. Adding a method to a class that does not have an interface doesn't because if one would have replaced the implementation, it needs to extend the original class and will inherit the new methods